### ISSUE TITLE
False-positive E1133 (not-an-iterable) for iterable context manager

### ISSUE THREAD

**Issue by tucked** [PyCQA/pylint#3233]
### Steps to reproduce
``` python
"""sscce.py"""

class Context:
    def __enter__(self):
        return self
    def __exit__(self, type_, exc, traceback):
        return False

class Iterable(Context):
    def __iter__(self):
        return self
    def __next__(self):
        return self

class Foo(Iterable):
    pass

with Foo() as foo:
    for s in foo:
        print(s)
```

Note: I built this SSCCE from a real-world error I got using `paramiko.SFTPFile`:
``` python
        with self._sftp.open(local_path, mode='r') as local_file:
            for line in local_file:  # triggers on local_file
                yield line.rstrip()
```
`SFTPFile` is like `Foo`. It inherits from `paramiko.BufferedFile` which is analogous to `Iterable`, and `BufferedFile` inherits from `paramiko.util.ClosingContextManager`.

### Current behavior
```
sscce.py:19:13: E1133: Non-iterable value foo is used in an iterating context (not-an-iterable)
```

### Expected behavior
(no errors)

### pylint --version output
```
pylint 2.4.3
astroid 2.3.3
Python 3.6.8 (default, Oct  7 2019, 12:59:55)
[GCC 8.3.0]
```

**Comment by PCManticore** [comment]
Thank you for the clear example, I can confirm the bug.

**Comment by cdce8p** [comment]
This issue seems to be fixed when testing with the latest pylint version.
If you still encounter issues, please report back and I'll reopen it.
